### PR TITLE
Vuln scan weekly

### PR DESCRIPTION
**Pull request by sehaartuc**
Updated trivy job checks out your code and finds all Docker images in the current GitHub repository.
For each Docker image found, it pulls the image and runs the Trivy vulnerability scanner on it.
The scanner checks for critical and high severity vulnerabilities in the operating system and libraries of each Docker image.

### CODE REVIEW

File: .github/workflows/weekly_vuln_scan.yml [hunk 1]
```diff
@@ -23,11 +22,19 @@ jobs:
...
+        run: |
+          echo "::set-output name=images::$(docker images --format '{{.Repository}}:{{.Tag}}')"
+
+      - name: Run Trivy vulnerability scanner on each Docker image
+        run: |
+          for image in ${{ steps.docker-images.outputs.images }}
+          do
+            echo "Scanning image: $image"
+            docker pull $image
+            trivy image --format table --exit-code 1 --ignore-unfixed --vuln-type os,library --severity CRITICAL,HIGH $image
```
[the remark this answers is not in the document]
Comment (sehaartuc) [reply .github/workflows/weekly_vuln_scan.yml]: We do not need to have it. It's removed on the last version.

File: .github/workflows/weekly_vuln_scan.yml [hunk 2]
```diff
@@ -23,11 +22,37 @@ jobs:
...
-        with:
-          format: "table"
-          exit-code: "1"
-          ignore-unfixed: true
-          vuln-type: "os,library"
-          severity: "CRITICAL,HIGH"
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Check if Trivy is installed
```
Comment (brunopacheco1) [inline .github/workflows/weekly_vuln_scan.yml:28]: question: why are you installing it manually? They already provide images to run trivy: `aquasecurity/trivy-action@master`

File: .github/workflows/weekly_vuln_scan.yml [hunk 3]
```diff
@@ -23,11 +22,19 @@ jobs:
...
-          ignore-unfixed: true
-          vuln-type: "os,library"
-          severity: "CRITICAL,HIGH"
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Find Docker images
+        id: docker-images
+        run: |
+          echo "::set-output name=images::$(docker images --format '{{.Repository}}:{{.Tag}}')"
```
[the remark this answers is not in the document]
Comment (sehaartuc) [reply .github/workflows/weekly_vuln_scan.yml:49]: It depends on counts of images but shouldn't be exhaustive. Additionally beneficial for security.

File: .github/workflows/weekly_vuln_scan.yml [hunk 4]
```diff
@@ -1,41 +1,35 @@
...
 
-      - name: Build Docker images
+      - name: Find Recent Docker Images
+        id: recent-images
         run: |
-          for Dockerfile in $(find . -name Dockerfile -mtime -7); do
-            image_name=$(dirname $Dockerfile | tr -d './')
-            docker build -t my-image-$image_name -f $Dockerfile .
-          done
+          FROM_DATE=$(date -d '7 days ago' '+%Y-%m-%d')
```
Comment (brunopacheco1) [inline .github/workflows/weekly_vuln_scan.yml:17]: The reference should not be time, but the last patch of the latest release or latest two, perhaps.

For example: if we have these branches `releases/v1.0`, `releases/v1.1`, and `releases/v2.0`, and theses tags `v1.0.0`, `v1.0.1`, `v1.1.0`, `v1.1.1`, `v1.1.2`, and `v2.0.0`, we should check only `v2.0.0` and `v1.1.2`.

Comment (sehaartuc) [reply .github/workflows/weekly_vuln_scan.yml]: I am updating it to detect latest two ones, thanks for the example.

**Comment by sehaartuc** [comment]
Hello, In order to list and fetch latest two tags, we need to put related Github Token into curl command. While implementing it in other repos, the only variable we need to change once is Github Token.

File: .github/workflows/weekly_vuln_scan.yml [hunk 5]
```diff
@@ -1,33 +1,50 @@
-# SPDX-FileCopyrightText: 2024 PNED G.I.E.
-# SPDX-FileContributor: Stichting Health-RI
-#
```
Comment (brunopacheco1) [inline .github/workflows/weekly_vuln_scan.yml:3]: important: you should put these comments back, otherwise the build will fail.

**Review by brunopacheco1 — APPROVED** [review]

File: .github/workflows/weekly_vuln_scan.yml (hunk 5 above)
Comment (sehaartuc) [reply .github/workflows/weekly_vuln_scan.yml:3]: Many thanks, I put related comments back in the yaml file. Now It needs only github token which can be defined in github secrets etc. It's up to you. Thanks again.

**Pull request merged**